scylla/src/transport/connection.rs

```diff
@@ -963,12 +963,18 @@ impl Connection {
         let (version_id,): (Uuid,) = self
             .query_single_page(LOCAL_VERSION)
             .await?
-            .rows
-            .ok_or(QueryError::ProtocolError("Version query returned not rows"))?
-            .into_typed::<(Uuid,)>()
-            .next()
-            .ok_or(QueryError::ProtocolError("Admin table returned empty rows"))?
-            .map_err(|_| QueryError::ProtocolError("Row is not uuid type as it should be"))?;
+            .single_row_typed()
+            .map_err(|err| match err {
+                SingleRowTypedError::RowsExpected(_) => {
+                    QueryError::ProtocolError("Version query returned not rows")
+                }
+                SingleRowTypedError::BadNumberOfRows(_) => {
+                    QueryError::ProtocolError("system.local query returned a wrong number of rows")
+                }
+                SingleRowTypedError::FromRowError(_) => {
+                    QueryError::ProtocolError("Row is not uuid type as it should be")
+                }
+            })?;
```

Comment:
I like this change, new version has shorter logic and separates logic from error handling, nice

Reply:
Hmmm... the behavior is different here as well. Previous version expected at least one row, new one expects exactly one row. Is this correct?

Reply:
Judging from [this snippet](https://github.com/scylladb/scylla-rust-driver/blob/main/scylla/src/transport/query_result.rs?plain=1#L114-L122), the previous behaviour expected exactly one row as well.